builders/build/eth-api/libraries/ethersrs.md

```diff
@@ -0,0 +1,644 @@
+---
+title: How to use Ethers.rs Ethereum Library
+description: Follow this tutorial to learn how to use the Ethereum EthersRS Library to send transactions and deploy Solidity smart contracts to Moonbeam via the Rust language.
+---
+
+# Ethers.rs Rust Library
+
+![Intro diagram](/images/builders/build/eth-api/libraries/ethersrs/ethers-rust-banner.png)
+
+## Introduction {: #introduction } 
+
+The [Ethers.rs](https://ethers.rs){target=_blank} library provides a set of tools to interact with Ethereum Nodes via the Rust programming language that works similar to [Ethers.js](/ethersjs){target=_blank}. Moonbeam has an Ethereum-like API available that is fully compatible with Ethereum-style JSON RPC invocations. Therefore, developers can leverage this compatibility and use the Ethers.rs library to interact with a Moonbeam node as if they were doing so on Ethereum. You can read more about how to use Ethers.rs on their [official crate documentation](https://docs.rs/ethers/latest/ethers/){target=_blank}.
+
+In this guide, you'll learn how to use the Ethers.rs library to send a transaction and deploy a contract on Moonbase Alpha. This guide can be adapted for [Moonbeam](/builders/get-started/networks/moonbeam/){target=_blank}, [Moonriver](/builders/get-started/networks/moonriver/){target=_blank}, or a [Moonbeam development node](/builders/get-started/networks/moonbeam-dev/){target=_blank}.
+
+## Checking Prerequisites {: #checking-prerequisites } 
+
+For the examples in this guide, you will need to have the following:
+
+ - An account with funds.
+  --8<-- 'text/faucet/faucet-list-item.md'
+ - 
+--8<-- 'text/common/endpoint-examples.md'
+ - Have [Rust installed](https://www.rust-lang.org/tools/install){target=_blank} on your device
+ - Have [solc installed](https://docs.soliditylang.org/en/v0.8.9/installing-solidity.html) on your device. Using [solc-select](https://github.com/crytic/solc-select){target=_blank} is recommended by the Ethers.rs package
+
+!!! note
+    The examples in this guide assumes you have a MacOS or Ubuntu 20.04-based environment and will need to be adapted accordingly for Windows.
```

Comment:
This can be replaced with the snippet /text/common/assumes-mac-or-ubuntu-env.md

Reply:
The snippet says 18.04, but I believe current versions of Rust require later versions.

Reply:
ahhh ok, missed that. thanks for clarifying!